### DISCUSSION TITLE
Change behavior of setting default value to $NOW

### DISCUSSION THREAD

**Discussion by JackMullinkosson** [directus/directus#22104] [Feature Requests]
### Summary

Right now you can set a value to default to $NOW, but it will default to the time that you saved this setting at, not the time that an item was created. $NOW in this case makes more sense to be the time the item was created, not the date/time the setting was saved. 

<img width="319" alt="Screenshot 2024-04-05 at 10 24 10 AM" src="https://github.com/directus/directus/assets/83920472/97793d91-7427-489d-b7ec-e855b275d859">

This similar discussion mentions a related issue, but it's unclear if the functionality I'm describing is planned: [https://github.com/directus/directus/discussions/11158](url)

### Motivation

This is for cases where you may want to manually set a date, and if no date was set manually then default it to the current date/time. This is not possible currently with "On Create" because if you have set it to current date/time, it will override any manual date that you set. 

### Detailed Design

No design change necessary - just change implementation of how $NOW is interpreted when set as default value. 

### Must Have:
- Ability to default to current time when created
- ability to manually set to any date as well

### Drawbacks

Not sure how this would be achieved given current libraries in use.

### Alternatives

Could have a dropdown option that is "Set to current time" instead of using the $NOW variable

### Adoption Strategy

Anybody who wants this functionality would likely have assumed that setting default value to $NOW would already have this result. The fact that it doesn't already is unexpected.

**Comment by br41nslug** [comment]
As described in the linked discussion:
> This exists as the "On Create" and "On Update" control values. There's nothing a default value allows you to do that the "On Create" doesn't achieve currently. The default value is saved to the database, which means there's no way to set it to "Current User", as the database isn't aware of what that user is! 🙂

https://github.com/directus/directus/discussions/11158#discussioncomment-5865601

Default values are handled by the database which does not have the context to do dynamic value replacement. To achieve the described behavior there are the "on Create" and "on Update" options which allow you to set the current date/time on creation or updates.

**Reply by JackMullinkosson** [reply]
the "on Create" option does not allow you to achieve the desired behavior. 

> This is not possible currently with "On Create" because if you have set it to current date/time, it will override any manual date that you set.

My feature request is to allow users to default to the current date/time, while still making it possible to manually set an item to a different date/time

**Reply by br41nslug** [reply]
This is unfortunately not something we can achieve at this time for the above referenced reasons. But is possible using a custom hook or flow to fill the current date/time if left empty.
